homeassistant/components/vicare/climate.py

```diff
@@ -101,21 +101,29 @@ def _build_entity(name, vicare_api, circuit, device_config, heating_type):
     return ViCareClimate(name, vicare_api, device_config, circuit, heating_type)
 
 
+def _get_circuits(vicare_api):
+    """Return the list of circuits."""
+    return vicare_api.circuits
+
+
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the ViCare climate platform."""
     name = VICARE_NAME
-
     entities = []
+    circuits = await hass.async_add_executor_job(
+        _get_circuits, hass.data[DOMAIN][config_entry.entry_id][VICARE_API]
+    )
 
     try:
-        for circuit in hass.data[DOMAIN][config_entry.entry_id][VICARE_API].circuits:
+        for circuit in circuits:
             suffix = ""
-            if len(hass.data[DOMAIN][config_entry.entry_id][VICARE_API].circuits) > 1:
+            if len(circuits) > 1:
                 suffix = f" {circuit.id}"
+
             entity = _build_entity(
```

Comment:
In a follow up, I'd remove `_build_entity` as it just instantiates the entity.

Reply:
OK, I'll create a separate PR